### PR TITLE
Refactor forms creation into forms repo

### PR DESCRIPTION
**Pull request by livborre**

### CODE REVIEW

File: src/forms/repositorySpec.js [hunk 1]
```diff
@@ -1,64 +1,130 @@
...
 
-describe("findForm function", function() {
-    it("fails if postgres throws an error while trying to find a form", async function() {
-        let db = {
-            query: jasmine.createSpy("db.query").and.callFake(async (query) => {
-                if(query.startsWith("SELECT * FROM forms")){
-                    throw new Error("postgres exploded while trying to find a form");
-                }
-            })
+fdescribe("forms repository", function() {
```
Comment (hypesystem) [inline src/forms/repositorySpec.js]: `fdescribe` committed here, should probably not be :wink:

File: src/forms/repositorySpec.js [hunk 2]
```diff
@@ -1,64 +1,130 @@
...
+                id = await formsRepository.create(hutId, formConfigs);
+            } catch(error) {
+                actualError = error; 
+            }
 
-        let db = {
-            query: jasmine.createSpy("db.query").and.callFake(async () => foundForm)
-        };
+            expect(actualError).toBe(null);
+            expect(db.query).toHaveBeenCalledWith(expectedFormsQuery,expectedFormsArguments(id));
```
Comment (hypesystem) [inline src/forms/repositorySpec.js]: You never assert that the `id` returned here is correct.

File: src/huts/repositorySpec.js [hunk 3]
```diff
@@ -153,27 +148,32 @@ describe("huts repository" , function() {
...
             expect(actualError).not.toBe(null);
             expect(db.query).toHaveBeenCalledWith(expectedHutQuery,expectedHutArguments(jasmine.any(String)));
         });
 
-        it("fails if postgress explodes upon inserting a form after having created a hut", async function() {
-            mockFailOnQuery(db, "INSERT INTO forms");
-            let hutsRepository = new HutsRepository(db)
+        it("fails if formsRepo explodes upon inserting a form after having created a hut", async function() {
+            formsRepo.create.and.callFake( async () => {
+                console.log("LOG")
```
Comment (hypesystem) [inline src/huts/repositorySpec.js]: Log statement snuck in, should be removed :smile:

File: spec/support/getErrorsFromRunningFunction.js [hunk 4]
```diff
@@ -0,0 +1,9 @@
+module.exports = async (runnable) => { 
+    let actualError = null;
+    try {
+        await runnable();
+    } catch(error) {
+        actualError = error;
+    }
+    return actualError;
+}
```
Comment (hypesystem) [inline spec/support/getErrorsFromRunningFunction.js]: git likes newlines at end of files

File: src/forms/repositorySpec.js [hunk 5]
```diff
@@ -1,64 +1,130 @@
...
 
-        expect(actualError).toBe(null);
-        expect(form).not.toEqual(undefined);
-    });
+            expect(actualError).not.toBe(null); 
+            expect(db.query).toHaveBeenCalledWith(expectedFormsQuery,expectedFormsArguments(jasmine.any(String)));
+        });
+    })
+    
+    describe("findForm function", function() {
```
Comment (hypesystem) [inline src/forms/repositorySpec.js]: `"find function"` should be sufficient: we are already in the forms repository test (inside a `describe("forms repository", ...`-block).

File: src/forms/repositorySpec.js (hunk 2 above)
Comment (livborre) [reply src/forms/repositorySpec.js]: hmm i'm not sure how to do that? What is a 'correct' id?
I do however assert that the returned id is also the one that was given as argumet to the database (when i call 'expectedFormsArguments(id)')

**Pull request merged**